Add a new module to logout inactive session after a configured delay

Cedric-Pigeon opened the pull request:
The goal of this PR is to set a validity on a user session.

The validity delay is set in a system parameter and its default value is 2 hours.
If the user do nothing during this delay, his session is automatically logged out and he must re-login.

lmignon commented on inactive_session_timeout/models/res_users.py:
```diff
@@ -0,0 +1,76 @@
+
+class res_users(models.Model):
+    _inherit = 'res.users'
+    _REMOTE_USER_ATTRIBUTE = 'HTTP_REMOTE_USER'
```
Not used

Cedric-Pigeon replied:
Removed...

hbrunn commented on inactive_session_timeout/README.rst:
```diff
@@ -0,0 +1,35 @@
+
+Two system parameters are available:
+
+* inactive_session_time_out_delay: validity of a session in secondes (default = 2 Hours)
```
seconds (without the second 'e')

Cedric-Pigeon replied:
Fixed

hbrunn commented on inactive_session_timeout/models/res_users.py:
```diff
@@ -0,0 +1,77 @@
+        session = request.session
+        session_store = root.session_store
+        registry = self.pool
+        if InactiveSessionTimeoutInstalled._name not in registry:
```
Why do you need this check if this is not a server wide module?

Cedric-Pigeon replied:
This was some remaining code due to a previous try. I removed it.

hbrunn commented on inactive_session_timeout/models/res_users.py:
```diff
@@ -0,0 +1,77 @@
+        try:
+            delay = int(param_model.get_param(cr, uid, DELAY_KEY, 7200))
+            urls = param_model.get_param(cr, uid, IGNORED_PATH_KEY,
+                                         '').split(',')
```
this causes 2 extra db requests per http requests, shouldn't you better cache this values?

Cedric-Pigeon replied:
Done.

hbrunn commented:
On a general note, if I needed this functionality I'd look what configuration parameters werkzeug has here. Did you do that? I honstely don't know if werkzeug provides anything, but I have the feeling this would be the right place to look

Commit:
Add a new module to logout inactive session after a configured delay

Cedric-Pigeon commented:
@hbrunn I tried to act earlier in the server on the http request but I didn't find any single entry point for all odoo requests. This is way I override the check method of res.user.

adrienpeiffer commented on inactive_session_timeout/README.rst:
```diff
@@ -0,0 +1,35 @@
+
+* inactive_session_time_out_delay: validity of a session in seconds (default = 2 Hours)
+* inactive_session_time_out_ignored_url: technical urls where the check does not occur 
+
```
Can you add bugt tracker section ?

adrienpeiffer commented on inactive_session_timeout/__openerp__.py:
```diff
@@ -0,0 +1,44 @@
+    'summary': """
+        This module disable all inactive sessions since a given delay""",
+
+    'author': "ACSONE SA/NV",
```
Add OCA as author like `ACSONE SA/NV,Odoo Community Association (OCA)`

lmignon commented on inactive_session_timeout/README.rst:
```diff
@@ -0,0 +1,35 @@
+Inactive Sessions Timeout
```
can you add the AGPL-3 badge

adrienpeiffer commented:
I restarted runbot build

pedrobaeza commented on inactive_session_timeout/models/ir_config_parameter.py:
```diff
@@ -0,0 +1,56 @@
+IGNORED_PATH_KEY = 'inactive_session_time_out_ignored_url'
+
+
+class ir_config_parameter(models.Model):
```
Use CamelCase for class name

pedrobaeza commented on inactive_session_timeout/models/res_users.py:
```diff
@@ -0,0 +1,62 @@
+from time import time
+
+
+class res_users(models.Model):
```
CamelCase

pedrobaeza commented:
Interesting module. Little remarks, but in general looks good

alejandrosantana commented on inactive_session_timeout/models/ir_config_parameter.py:
```diff
@@ -0,0 +1,56 @@
+
+from openerp import tools
+from openerp import SUPERUSER_ID
+
```
Wouldn't it be OK in just one line?:
`from openerp import api, models, SUPERUSER_ID, tools`

alejandrosantana commented:
Wouldn't it be enough overriding `openerp/http.py` and changing the hardcoded value 60*60*24*7 with the value of your parameter? Or also helpful?
https://github.com/odoo/odoo/blob/8.0/openerp/http.py#L1164

``` python

def session_gc(session_store):
    if random.random() < 0.001:
        # we keep session one week
        last_week = time.time() - 60*60*24*7
        for fname in os.listdir(session_store.path):
            path = os.path.join(session_store.path, fname)
            try:
                if os.path.getmtime(path) < last_week:
                    os.unlink(path)
            except OSError:
                pass
```

Cedric-Pigeon commented:
@lmignon Done. Thanks for reviewing.
@adrienpeiffer Done. Thanks for reviewing.
@pedrobaeza Done. Thanks for reviewing.
@alejandrosantana Done. Thanks for reviewing.

nhomar commented:
:+1:

pedrobaeza commented:
:+1:

laetitia-gangloff commented on inactive_session_timeout/README.rst:
```diff
@@ -0,0 +1,46 @@
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/{project_repo}/issues>`_.
```
replace value in {}

Commit:
Commit concerns:
  - Add AGPL-3 badge on README.rst
  - Insert Bug tracker informations in README.rst
  - Import from same package on same line
  - Use CamelCase for class names

Cedric-Pigeon commented:
@laetitia-gangloff Done, thanks.

laetitia-gangloff commented:
:+1:

alejandrosantana commented:
@Cedric-Pigeon Nice module. I am testing it functionally, and looks fine.

:+1:
